### PR TITLE
Update IDE plugin to 2024.2

### PR DESCRIPTION
**Pull request by chippmann**
This updates our IDE plugin to 2024.2 and updated the gradle setup to use the new [intellij platform gradle plugin](https://plugins.jetbrains.com/docs/intellij/tools-intellij-platform-gradle-plugin.html)

### CODE REVIEW

**Review by CedNaru — APPROVED** [review]

File: kt/plugins/godot-intellij-plugin/src/main/kotlin/godot/intellij/plugin/gradle/GodotKotlinJvmSettings.kt [hunk 1]
```diff
@@ -13,6 +13,7 @@ import com.intellij.ui.EditorNotifications
 import godot.intellij.plugin.GodotPluginBundle
 import godot.intellij.plugin.notification.SettingsFetchingNotification
 import godot.plugins.common.GodotKotlinJvmPropertiesFile
+import godot.plugins.common.gradle.action.FetchProjectModelsBuildAction
```
Comment (piiertho) [inline kt/plugins/godot-intellij-plugin/src/main/kotlin/godot/intellij/plugin/gradle/GodotKotlinJvmSettings.kt:16]: why is this added ?

**Review by piiertho — APPROVED** [review]

File: kt/plugins/godot-intellij-plugin/src/main/kotlin/godot/intellij/plugin/gradle/GodotKotlinJvmSettings.kt (hunk 1 above)
Comment (chippmann) [reply kt/plugins/godot-intellij-plugin/src/main/kotlin/godot/intellij/plugin/gradle/GodotKotlinJvmSettings.kt:16]: because we moved that class from the same package to the common module instead. so an import is needed
